SDK/src/NDK/Widgets/CheckboxWidget.cpp

```diff
@@ -0,0 +1,194 @@
+		                                                                   origin.y + checkboxSize.y / 2.f - checkboxTextBox.y / 2.f);
+
+		Nz::Vector3f textBox = m_textSprite->GetBoundingVolume().obb.localBox.GetLengths();
+		m_textEntity->GetComponent<NodeComponent>().SetPosition(origin.x + checkboxSize.x + (m_adaptativeMargin ? checkboxSize.x / 2.f : m_textMargin),
+																origin.y + checkboxSize.y / 2.f - textBox.y / 2.f);
+	}
+
+	void CheckboxWidget::OnMouseButtonRelease(int x, int y, Nz::Mouse::Button button)
+	{
+		if (button == Nz::Mouse::Left && ContainsCheckbox(x, y) && IsCheckboxEnabled())
+		{
+			SetNextState();
+			OnStateChanged(this);
+		}
+	}
+
+
+	void CheckboxWidget::UpdateCheckboxSprite()
+	{
+		if (m_checkboxEnabled)
+			m_checkboxSprite->SetTexture(m_checkbox, false);
+
+		else
+			m_checkboxSprite->SetTexture(m_disabledCheckbox, false);
+
+
+		if (m_state == CheckboxState_Unchecked)
+		{
+			m_checkboxContentEntity->Enable(false);
+			return;
+		}
+
+		else if (m_state == CheckboxState_Checked)
+		{
+			m_checkboxContentEntity->Enable();
+			m_checkboxContentSprite->SetTexture(m_checkboxContentChecked, false);
+		}
+
+		else
+		{
+			m_checkboxContentEntity->Enable();
+			m_checkboxContentSprite->SetTexture(m_checkboxContentTristate, false);
+		}
+
+		m_checkboxContentSprite->SetColor(m_checkboxEnabled ? m_mainColor : m_disabledMainColor);
+	}
+
+
+	void CheckboxWidget::CreateCheckboxTextures()
+	{
+		Nz::Vector2ui checkboxSize = { 32u, 32u };
+		Nz::Vector2ui borderSize = checkboxSize / static_cast<unsigned>(m_borderScale);
+		SetCheckboxSize(Nz::Vector2f { static_cast<float>(checkboxSize.x), static_cast<float>(checkboxSize.y) });
+
+
+		Nz::Image checkbox;
+		checkbox.Create(Nz::ImageType_2D, Nz::PixelFormatType_L8, checkboxSize.x, checkboxSize.y);
+
+		checkbox.Fill(m_mainColor, Nz::Rectui { checkboxSize.x, checkboxSize.y });
+		checkbox.Fill(m_backgroundColor, Nz::Rectui { borderSize.x, borderSize.y, checkboxSize.x - (borderSize.x * 2), checkboxSize.y - (borderSize.y * 2) });
+
+		m_checkbox = Nz::Texture::New(checkbox);
+
+		Nz::Image disabledCheckbox;
+		disabledCheckbox.Create(Nz::ImageType_2D, Nz::PixelFormatType_L8, checkboxSize.x, checkboxSize.y);
+
+		disabledCheckbox.Fill(m_disabledMainColor, Nz::Rectui { checkboxSize.x, checkboxSize.y });
+		disabledCheckbox.Fill(m_disabledBackgroundColor, Nz::Rectui { borderSize.x, borderSize.y, checkboxSize.x - (borderSize.x * 2), checkboxSize.y - (borderSize.y * 2) });
+
+		m_disabledCheckbox = Nz::Texture::New(disabledCheckbox);
+
+
+		Nz::Image tristate;
+		tristate.Create(Nz::ImageType_2D, Nz::PixelFormatType_L8, 1u, 1u);
+		tristate.Fill(Nz::Color::Black);
+		m_checkboxContentTristate = Nz::Texture::New(tristate);
+
+
+		const Nz::UInt8 r_checkmark[] =
+		{
+			#include <NDK/Resources/checkmark.png.h>
+		};
+
+		m_checkboxContentChecked = Nz::Texture::New();
+		m_checkboxContentChecked->LoadFromMemory(r_checkmark, sizeof(r_checkmark) / sizeof(r_checkmark[0]));
```

Comment:
You're having every checkbox widget create its own set of textures, you should do it once for all (with some initialization methods)